src/backend/joanie/payment/models.py

```diff
@@ -344,12 +344,31 @@ def save(self, *args, **kwargs):
         super().save(*args, **kwargs)
 
 
+class CreditCardManager(models.Manager):
+    """Custom manager for `CreditCard` model"""
+
+    def get_card_for_owner(self, token, user_id):
+        """
+        Retrieve a main credit card for a given owner.
+        This method attempts to retrieve a credit card that matches the given `token` and
+        `user_id`, and is marked as the main card. If no such card exists, it returns None.
+        """
+        try:
+            return self.get(token=token, owner_id=user_id, is_main=True)
+        except self.model.DoesNotExist:
```

Comment:
Why catching this exception ? You should not

Reply:
I will take the try except block here.  May I know why it is not suitable to catch the exception here ? I thought it would have been a good idea with a GET queryset.